src/components/spreadsheet/utils/equipment-table-utils.ts

```diff
@@ -80,6 +81,93 @@ export const updateGeneratorCells = (params: CellEditingStoppedEvent) => {
+    switch (colId) {
+        case 'ratioTapChanger.regulationType':
+            if (ratioRegulationTypeText === REGULATION_TYPES.DISTANT.id) {
+                // This solve the problem of required on RatioRegulatingTerminal when we set regulationType to distant
+                const regulatingTerminalGenerator =
+                    getTapChangerRegulationTerminalValue(
+                        previousData?.ratioTapChanger
+                    );
+                rowNode.setDataValue(
+                    'RatioRegulatingTerminal',
+                    regulatingTerminalGenerator
+                );
+            }
+            params.api.flashCells({
+                rowNodes: [rowNode],
+                columns: [
+                    'ratioTapChanger.regulationType',
+                    'RatioRegulatingTerminal',
+                ],
+            });
+            break;
+
+        case 'phaseTapChanger.regulationType':
+            if (phaseRegulationTypeText === REGULATION_TYPES.DISTANT.id) {
+                // This solve the problem of required on PhaseRegulatingTerminal when we set regulationType to distant
+                const regulatingTerminalGenerator =
+                    getTapChangerRegulationTerminalValue(
+                        previousData?.phaseTapChanger
+                    );
+                rowNode.setDataValue(
+                    'PhaseRegulatingTerminal',
+                    regulatingTerminalGenerator
+                );
+            }
+            params.api.flashCells({
+                rowNodes: [rowNode],
+                columns: [
+                    'phaseTapChanger.regulationType',
+                    'PhaseRegulatingTerminal',
+                ],
+            });
+            break;
+
+        case 'RatioRegulatingTerminal':
+            const RatioRegulatingTerminal = params.data.RatioRegulatingTerminal;
+            if (RatioRegulatingTerminal) {
+                params.data.ratioTapChanger.regulatingTerminalVlId =
+                    params.context.dynamicValidation.ratioTapChanger.regulatingTerminalVlId;
+                params.data.ratioTapChanger.regulatingTerminalConnectableId =
+                    params.context.dynamicValidation.ratioTapChanger.regulatingTerminalConnectableId;
+                params.data.ratioTapChanger.regulatingTerminalConnectableType =
+                    params.context.dynamicValidation.ratioTapChanger.regulatingTerminalConnectableType;
+            } else {
+                params.data.regulatingTerminalVlId = undefined;
+                params.data.regulatingTerminalConnectableId = undefined;
+                params.data.regulatingTerminalConnectableType = undefined;
+            }
+            break;
+
+        case 'PhaseRegulatingTerminal':
+            const PhaseRegulatingTerminal = params.data.PhaseRegulatingTerminal;
+            if (PhaseRegulatingTerminal) {
+                params.data.phaseTapChanger.regulatingTerminalVlId =
```

Comment:
There is an error when you modify only the "regulated terminal" of a TWT that is already in "DISTANT" regulation